### PR TITLE
Storage Implementation

### PR DESCRIPTION
**Pull request by Bendistocratic**
Added saveLocationCommand to change storage location. Incomplete

### CODE REVIEW

**Comment by Bendistocratic** [comment]
1. There is no test cases as of now.
2. There are some issues with the file path finding. (It creates a new folder instead of throwing an exception)

**Comment by xpdavid** [comment]
@Bendistocratic Should be OK I think. If the path doesn't exist, you create a new one (no need to throw exception). This is done by many software.

File: src/main/java/seedu/doerList/logic/LogicManager.java [hunk 1]
```diff
@@ -22,17 +22,19 @@
...
         this.parser = new Parser();
+        this.storage = storage;
     }
 
     @Override
     public CommandResult execute(String commandText) {
         logger.info("----------------[USER COMMAND][" + commandText + "]");
         Command command = parser.parseCommand(commandText);
-        command.setData(model);
+        command.setData(model, storage);
```
Comment (xpdavid) [inline src/main/java/seedu/doerList/logic/LogicManager.java]: Storage should not depends on any component. Using event design instead.

File: src/main/java/seedu/doerList/model/ModelManager.java [hunk 2]
```diff
@@ -98,6 +99,12 @@ public synchronized void unmarkTask(ReadOnlyTask task) throws TaskNotFoundExcept
         doerList.unmarkTask(task);
         indicateDoerListChanged();
     }
+    
+    @Override
+    public synchronized void changeSaveLocation(String saveLocation, String fileName) {
```
Comment (xpdavid) [inline src/main/java/seedu/doerList/model/ModelManager.java]: It is not Model Manager's responsibility to change the save location!

File: .project [hunk 3]
```diff
@@ -1,6 +1,6 @@
 <?xml version="1.0" encoding="UTF-8"?>
 <projectDescription>
-	<name>Do-erList</name>
+	<name>DoerList</name>
```
Comment (xpdavid) [inline .project]: Change this two files back!

Comment (xpdavid) [inline .project]: My phone runs out of power. :(

File: .settings/org.eclipse.buildship.core.prefs [hunk 4]
```diff
@@ -1,6 +1,7 @@
 build.commands=org.eclipse.jdt.core.javabuilder
 connection.arguments=
 connection.gradle.distribution=GRADLE_DISTRIBUTION(WRAPPER)
+connection.gradle.user.home=null
```
Comment (xpdavid) [inline .settings/org.eclipse.buildship.core.prefs]: Change back

File: src/main/java/seedu/doerList/storage/StorageManager.java [hunk 5]
```diff
@@ -21,7 +30,7 @@
 public class StorageManager extends ComponentManager implements Storage {
 
     private static final Logger logger = LogsCenter.getLogger(StorageManager.class);
-    private XmlDoerListStorage doerListStorage;
+    private static XmlDoerListStorage doerListStorage;
```
Comment (xpdavid) [inline src/main/java/seedu/doerList/storage/StorageManager.java]: You can apply `singleton` design pattern here.
Making is static is not recommended.

File: src/test/java/guitests/SaveLocationCommandTest.java [hunk 6]
```diff
@@ -0,0 +1,33 @@
...
+    @Test
+    public void change_save_location_successful() throws IllegalValueException {
+        
+        commandBox.runCommand("saveto /st " + TestUtil.SANDBOX_FOLDER + " /n " + "sampleData.xml");
+        String expectedSaveLocation = getDataFileLocation();
+        assertSaveLocationSuccess(TestUtil.SANDBOX_FOLDER + "\\sampleData.xml", expectedSaveLocation);
+        
+        // invalid command
+        commandBox.runCommand("savesto /st C:\\Documents /n MyList");
+        assertResultMessage(Messages.MESSAGE_UNKNOWN_COMMAND);
```
Comment (xpdavid) [inline src/test/java/guitests/SaveLocationCommandTest.java]: Need to validate the destination file has exactly the same content as the data here.

File: src/test/java/seedu/doerList/logic/LogicManagerTest.java [hunk 7]
```diff
@@ -106,6 +110,22 @@ public void execute_invalid() throws Exception {
...
+     * also checks if the new save location of data is correct.
+     */
+    private void assertCommandBehavior(String inputCommand, 
+                                       String expectedMessage,
+                                       String expectedFilePath) throws Exception {
+        
+        CommandResult result = logic.execute(inputCommand);
+        assertEquals(expectedMessage, result.feedbackToUser);
+        
+        File file = new File(expectedFilePath);
```
Comment (xpdavid) [inline src/test/java/seedu/doerList/logic/LogicManagerTest.java]: Need to validate the content!

File: src/test/java/guitests/SaveLocationCommandTest.java [hunk 8]
```diff
@@ -0,0 +1,33 @@
...
+
+    @Test
+    public void change_save_location_successful() throws IllegalValueException {
+        
+        commandBox.runCommand("saveto /st " + TestUtil.SANDBOX_FOLDER + " /n " + "sampleData.xml");
+        String expectedSaveLocation = getDataFileLocation();
+        assertSaveLocationSuccess(TestUtil.SANDBOX_FOLDER + "\\sampleData.xml", expectedSaveLocation);
+        
+        // invalid command
+        commandBox.runCommand("savesto /st C:\\Documents /n MyList");
```
Comment (xpdavid) [inline src/test/java/guitests/SaveLocationCommandTest.java]: The path `C:\\Documents` actually depends on system.
The Tavic CI test runs in linux. Definitely will fail.
Pls check the type of OS and supply correct path.

**Comment by Bendistocratic** [comment]
I'm not sure why Travis is unable to get the tempfolder. I tried printing it out on windows and it works.

**Pull request closed without merging**